### PR TITLE
MBS-10309: Updated URL cleanup for musik-sammler

### PR DESCRIPTION
**Pull request by SothoTalKer**
Since musik-sammler changed the URL format for releases a tiny bit, I have updated the cleanup URL along with the suggested changes by chaban. See https://tickets.metabrainz.org/browse/MBS-10309

### CODE REVIEW

**Comment by SothoTalKer** [comment]
Adressed 1. but could not reproduce 2.

**Comment by SothoTalKer** [comment]
That should do the trick.
Artists do not have their ID appended, so those URLs are handled differently. Both name and ID will work, along with URLs that contain a %. Separare regex then.
Updated the tests with some of those URLs, too. This should likely cover all the edge cases. :)

File: root/static/scripts/edit/URLCleanup.js [hunk 1]
```diff
@@ -1539,7 +1539,9 @@ const CLEANUPS = {
     match: [new RegExp('^(https?://)?(www\\.)?musik-sammler\\.de/', 'i')],
     type: LINK_TYPES.otherdatabases,
     clean: function (url) {
-      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?musik-sammler\.de\/(album|artist|media)\/([0-9a-z-]+)(?:[\/?#].*)?$/, 'https://www.musik-sammler.de/$1/$2/');
+      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?musik-sammler\.de\/artist\/?([0-9a-zA-Z-%]+)(?:[\/?#].*)?$/, 'https://www.musik-sammler.de/artist/$1/');
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js]: Slash seems to be mandatory here.
```suggestion
      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?musik-sammler\.de\/artist\/([0-9a-zA-Z-%]+)(?:[\/?#].*)?$/, 'https://www.musik-sammler.de/artist/$1/');
```

File: root/static/scripts/edit/URLCleanup.js [hunk 2]
```diff
@@ -1539,7 +1539,9 @@ const CLEANUPS = {
...
+      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?musik-sammler\.de\/album\/(?:[^\/]+-(?=[\d+\/]))?(\d+)(?:[\/?#].*)?$/, 'https://www.musik-sammler.de/album/$1/');
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js]: Character `+` is not expected here.

```suggestion
      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?musik-sammler\.de\/album\/(?:[^\/]+-(?=[\d\/]))?(\d+)(?:[\/?#].*)?$/, 'https://www.musik-sammler.de/album/$1/');
```

File: root/static/scripts/edit/URLCleanup.js [hunk 3]
```diff
@@ -1539,7 +1539,9 @@ const CLEANUPS = {
...
+      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?musik-sammler\.de\/(?:media|release)\/(?:[^\/]+-(?=[\d+\/]))?(\d+)(?:[\/?#].*)?$/, 'https://www.musik-sammler.de/release/$1/');
```
Comment (yvanzo) [inline root/static/scripts/edit/URLCleanup.js]: Same as above.

```suggestion
      url = url.replace(/^(?:https?:\/\/)?(?:www\.)?musik-sammler\.de\/(?:media|release)\/(?:[^\/]+-(?=[\d\/]))?(\d+)(?:[\/?#].*)?$/, 'https://www.musik-sammler.de/release/$1/');
```

**Comment by yvanzo** [comment]
Thank you, I took the liberty to complete your commit message with issues we discussed here.
